test/forge/invariant/MorphoInvariantTest.sol

```diff
@@ -62,15 +62,17 @@ contract MorphoInvariantTest is InvariantTest {
     }
 
     modifier authorized(address onBehalf) {
-        if (onBehalf != msg.sender) {
+        if (onBehalf != msg.sender && !morpho.isAuthorized(onBehalf, msg.sender)) {
             vm.prank(onBehalf);
             morpho.setAuthorization(msg.sender, true);
         }
 
         _;
 
-        vm.prank(onBehalf);
-        morpho.setAuthorization(msg.sender, false);
+        if (morpho.isAuthorized(onBehalf, msg.sender)) {
```

Comment:
Isn't `msg.sender` necessarily authorized to manage `onBehalf` at this point?

Reply:
No: Morpho's state could be modified inside `_;`
Perhaps is it worth commenting